sources/engine/ChessEngine.h

```diff
@@ -56,6 +56,8 @@ class ChessEngine
         Piece::Type     promotionType = Piece::Type::QUEEN;
         Piece::Color    turn;
 
+        int fmCounter; //"fifty-moves"
```

Comment:
It always will be unsigned, so `int8_t` ?

Reply:
`moveCounter` is better than `fmCounter` imo

Reply:
Based on fen format, `moveCounter` should be used to describe another variable.